packages/react/src/components/Animation/Animation.tsx

```diff
@@ -139,92 +142,158 @@ export interface AnimationProps extends StyledComponentProps {
-    name: PropTypes.string,
-    delay: PropTypes.string,
-    direction: PropTypes.string,
-    duration: PropTypes.string,
-    fillMode: PropTypes.string,
-    iterationCount: PropTypes.string,
-    keyframeParams: PropTypes.object,
-    playState: PropTypes.string,
-    timingFunction: PropTypes.string,
-    visible: PropTypes.bool,
-    appear: PropTypes.bool,
-    mountOnEnter: PropTypes.bool,
-    unmountOnExit: PropTypes.bool,
-    timeout: PropTypes.oneOfType([
-      PropTypes.number,
-      PropTypes.shape({
-        appear: PropTypes.number,
-        enter: PropTypes.number,
-        exit: PropTypes.number,
-      }),
-    ]),
-    onEnter: PropTypes.func,
-    onEntering: PropTypes.func,
-    onEntered: PropTypes.func,
-    onExit: PropTypes.func,
-    onExiting: PropTypes.func,
-    onExited: PropTypes.func,
-  }
-
-  static defaultProps = {
-    timeout: 0,
-  }
-
-  handleAnimationEvent = (
+const Animation: React.FC<AnimationProps> & {
+  className: string
+  handledProps: (keyof AnimationProps)[]
+} = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext)
+  const { setStart, setEnd } = useTelemetry(Animation.displayName, context.telemetry)
+  setStart()
+
+  const {
+    appear,
+    children,
+    className,
+    delay,
+    direction,
+    duration,
+    fillMode,
+    iterationCount,
+    keyframeParams,
+    mountOnEnter,
+    name,
+    playState,
+    timeout,
+    timingFunction,
+    visible,
+    unmountOnExit,
+  } = props
+
+  const handleAnimationEvent = (
     event: 'onEnter' | 'onEntering' | 'onEntered' | 'onExit' | 'onExiting' | 'onExited',
   ) => () => {
-    _.invoke(this.props, event, null, this.props)
+    _.invoke(props, event, null, props)
   }
 
-  renderComponent({ ElementType, classes, unhandledProps }) {
-    const { children, mountOnEnter, unmountOnExit, timeout, appear, visible } = this.props
-
-    const isChildrenFunction = typeof children === 'function'
-
-    const child =
-      childrenExist(children) &&
-      !isChildrenFunction &&
-      (React.Children.only(children) as React.ReactElement<any>)
-
-    return (
+  const { classes } = React.useMemo(() => {
+    const animation: ComponentAnimationProp = {
+      name,
+      keyframeParams,
+      duration,
+      delay,
+      iterationCount,
+      direction,
+      fillMode,
+      playState,
+      timingFunction,
+    }
+
+    return getStyles({
+      className: Animation.className,
+      displayName: Animation.displayName,
+      props: {
+        className,
+        styles: createAnimationStyles(animation, context.theme),
+      },
+
+      disableAnimations: context.disableAnimations,
+      renderer: context.renderer,
+      rtl: context.rtl,
+      saveDebug: _.noop,
+      theme: context.theme,
+      _internal_resolvedComponentVariables: context._internal_resolvedComponentVariables,
+    })
+  }, [
+    className,
+    context,
```

Comment:
Are we using everything from the context? Should we maybe list the things used from the context

Reply:
```
     disableAnimations: context.disableAnimations,
      renderer: context.renderer,
      rtl: context.rtl,
      theme: context.theme,
```

Almost everything... So I don't see any benefits in destructing it...